core/store/src/flat/delta.rs

```diff
@@ -0,0 +1,204 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+
+use near_primitives::state::ValueRef;
+use near_primitives::types::{RawStateChangesWithTrieKey, ShardId};
+use std::collections::HashMap;
+
+#[cfg(feature = "protocol_feature_flat_state")]
+use super::store_helper;
+use crate::{CryptoHash, StoreUpdate};
+
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct KeyForFlatStateDelta {
+    pub shard_id: ShardId,
+    pub block_hash: CryptoHash,
+}
+
+/// Delta of the state for some shard and block, stores mapping from keys to value refs or None, if key was removed in
+/// this block.
+#[derive(BorshSerialize, BorshDeserialize, Clone, Default, Debug, PartialEq, Eq)]
+pub struct FlatStateDelta(pub(crate) HashMap<Vec<u8>, Option<ValueRef>>);
```

Comment:
We didn't discuss that but it probably makes sense to rename it to `FlatStorageDelta` for consistency with other structs.

Reply:
decided to keep it as `FlatStateDelta` to be consistent with `FlatState`/`FlatStateDeltas` db columns